packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx

```diff
@@ -0,0 +1,189 @@
+    if (span.hasChildren) {
+      const Children = spans
+        .filter(span2 =>
+          span2.references.some(
+            reference => reference.refType === 'CHILD_OF' && reference.spanID === span.spanID
+          )
+        )
+        .sort((a, b) => b.startTime + b.duration - (a.startTime + a.duration))
+        .map(span2 => span2.spanID);
+      refinedSpanData.push({ ...span, childSpanIds: Children });
+    } else {
+      refinedSpanData.push({ ...span, childSpanIds: [] });
+    }
+  });
+  return refinedSpanData;
+};
+
+// This function finds LFC of a current span which is less than the spawn event
+export const findLastFinishingChildSpanId = (
+  traceData: Trace,
+  currentSpan: Span,
+  spawnTime?: number
+): string | undefined => {
+  if (spawnTime) {
+    return currentSpan.childSpanIds.find(each =>
+      traceData.spans.some(span => span.spanID === each && span.startTime + span.duration < spawnTime)
+    );
+  }
+  return currentSpan.childSpanIds[0];
+};
+
+// This function turncates/drops the overflowing child spans
+export const sanitizeOverFlowingChildren = (spans: Span[]): Span[] => {
+  const sanitizedSpanData: Span[] = [];
+  const droppedSpans: Span[] = [];
+  const refinedSantitizedSpanData: Span[] = [];
+
+  spans.forEach(span => {
+    if (span.references.length) {
+      const parentSpan = span.references.filter(ref => ref.refType === 'CHILD_OF')[0].span!;
+      const childEndTime = span.startTime + span.duration;
+      const parentEndTime = parentSpan.startTime + parentSpan.duration;
+      if (span.startTime >= parentSpan.startTime && childEndTime <= parentEndTime) {
+        // case 1: everything looks good
+        // |----parent----|
+        //   |----child--|
+        sanitizedSpanData.push(span);
+      } else if (
+        span.startTime < parentSpan.startTime &&
+        childEndTime <= parentEndTime &&
+        childEndTime > parentSpan.startTime
+      ) {
+        // case 2: child start before parent, truncate is needed
+        //      |----parent----|
+        //   |----child--|
+        sanitizedSpanData.push({
+          ...span,
+          startTime: parentSpan.startTime,
+          duration: childEndTime - parentSpan.startTime,
+        });
+      } else if (
+        span.startTime >= parentSpan.startTime &&
+        childEndTime > parentEndTime &&
+        span.startTime < parentEndTime
+      ) {
+        // case 3: child end after parent, truncate is needed
+        //      |----parent----|
+        //              |----child--|
+        sanitizedSpanData.push({ ...span, duration: parentEndTime - span.startTime });
+      } else {
+        // case 4: child outside of parent range =>  drop the child span
+        //      |----parent----|
+        //                        |----child--|
+        // or
+        //                      |----parent----|
+        //       |----child--|
+        droppedSpans.push({ ...span });
+      }
+    } else {
+      sanitizedSpanData.push(span);
```

Comment:
If we know that the value of `span.references.length` is undefined, let's make an early return by handling this case at first, rather than handling it at the start. 

For e.g.
```
if (someValidCondition){
// complicated business logic
} else {
 // small business logic
}
```
you can replace it with
```
if(not someValidCondition){
// smallBusinessLogic
}

// complicatedBusinessLogic
```